traveler_app/src/main/java/com/guestlogix/traveler/activities/SettingsActivity.java

```diff
@@ -0,0 +1,88 @@
+package com.guestlogix.traveler.activities;
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.guestlogix.traveler.R;
+import com.guestlogix.travelercorekit.utilities.TravelerPrefs;
+
+import static com.guestlogix.travelercorekit.utilities.UrlHelper.isValid;
+
+public class SettingsActivity extends AppCompatActivity {
+
+    private TravelerPrefs travelerPrefs = TravelerPrefs.getInstance(this);
+
+    private TextView sdkEndpointEditText;
+    private TextView authEndpointEditText;
+    private Button updateSDKEndpointButton;
+    private Button resetSDKEndpointButton;
+    private Button updateAuthEndpointButton;
+    private Button resetAuthEndpointButton;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_settings);
+
+        sdkEndpointEditText = findViewById(R.id.newSDkEndpointEditText);
+        authEndpointEditText = findViewById(R.id.newAuthEndpointEditText);
+        updateSDKEndpointButton = findViewById(R.id.updateSDkEndpointButton);
+        resetSDKEndpointButton = findViewById(R.id.resetSDkEndpointButton);
+        updateAuthEndpointButton = findViewById(R.id.updateAuthEndpointButton);
+        resetAuthEndpointButton = findViewById(R.id.resetAuthEndpointButton);
+
+        updateSDKEndpointButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                String newSDKEndpoint = sdkEndpointEditText.getText().toString();
+                if (isValid(newSDKEndpoint)) {
+                    travelerPrefs.save("TRAVELER_SDK_ENDPOINT", newSDKEndpoint);
+                    Toast.makeText(getApplicationContext(),
+                            String.format("Traveler SDK endpoint has been set to %s", newSDKEndpoint),
+                            Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+
+        resetSDKEndpointButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                travelerPrefs.remove("TRAVELER_SDK_ENDPOINT");
+                Toast.makeText(getApplicationContext(),
+                        "Traveler SDK endpoint has been reset",
```

Comment:
several strings in this activity can be moved to strings.xml

Reply:
removed